cow_instrument/SourceSampleDetectorPathFactory.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SOURCE_SAMPLE_DETECTOR_PATH_FACTORY_TEST_H
+#define SOURCE_SAMPLE_DETECTOR_PATH_FACTORY_TEST_H
+
+#include "PathFactory.h"
+#include "Path.h"
+#include <vector>
+
+/**
+ * All Paths generated for all detectors in this PathFactory type
+ * will go source->sample->detector.
+ */
+template <typename InstTree>
+class SourceSampleDetectorPathFactory : public PathFactory<InstTree> {
+
+public:
+  Paths *createL2(const InstTree &instrument) const;
```

Comment:
I don't need to covariance on the return type, so I'll change these to `unique_ptrs` that way they can just be moved when the `DetectoInfo` stores them as `shared_ptr`.